src/pydicom/pixels/utils.py

```diff
@@ -2098,3 +2170,246 @@ def unpack_bits(src: bytes, as_array: bool = True) -> "np.ndarray | bytes":
         raise ValueError("unpack_bits() requires NumPy if 'as_array = True'")
 
     return b"".join(map(_UNPACK_LUT.__getitem__, src))
+
+
+def get_packed_frame(
+    src: bytes,
+    index: int,
+    frame_length: int,
+    pad: bool = True,
+) -> bytes:
+    """Retrieve a single bit-packed frame from a bit-packed buffer.
+
+    This should only be used for single bit data (Bits Allocated = 1) in a
+    "bit-packed" representation (one byte containing 8 pixels). The pixel data
+    of such images may have pixel data from multiple frames within a single
+    byte.
+
+    Parameters
+    ----------
+    src : bytes
+        The native bit-packed representation of a full PixelData element,
+        potentially including multiple frames.
+    index : int
+        Zero-based index of the frame to extract.
+    frame_length : int
+        Number of pixels in each frame.
+    pad : bool
+        Whether to zero-pad the resulting bytes to a multiple of 2 bytes.
+
+    Returns
+    -------
+    bytes | bytearray
+        Bit-packed representation of the single frame requested.
+
+    Note
+    ----
+    In certain situations, this function will run faster and use less memory if
+    NumPy is installed. Otherwise, it will fall back to a slower method using
+    native Python functions.
+    """
+    # Pixel index of first pixel in requested frame
+    pixel_start = index * frame_length
+
+    # Pixel index one beyond the last pixel in the requested frame
+    pixel_end = pixel_start + frame_length
+
+    # Index of first byte containing pixels from the requested frame
+    byte_start = math.floor(pixel_start / 8)
+
+    # Index of the byte one beyond that containing last pixel in the requested
+    # frame
+    byte_end = math.floor((pixel_end - 1) / 8) + 1
+
+    # Mask for the final byte of the frame
+    final_byte_mask = [
+        b"\xff",  # should never be used
+        b"\x01",
+        b"\x03",
+        b"\x07",
+        b"\x0f",
+        b"\x1f",
+        b"\x3f",
+        b"\x7f",
+    ][frame_length % 8]
```

Comment:
Why not just use the integer values?

Reply:
Thanks, this is neater

Addressed in [e9ea3de](https://github.com/pydicom/pydicom/pull/2216/commits/e9ea3deb538f6786f5a80e0e826d7aaf7a40678c)